nexus/db-queries/src/db/datastore/rack.rs

```diff
@@ -1373,166 +1373,198 @@ mod test {
             SledUuid::from_untyped_uuid(sled1.id()),
             BlueprintZonesConfig {
                 generation: Generation::new().next(),
-                zones: vec![
-                    BlueprintZoneConfig {
-                        disposition: BlueprintZoneDisposition::InService,
-                        id: external_dns_id,
-                        filesystem_pool: Some(dataset.pool_name.clone()),
-                        zone_type: BlueprintZoneType::ExternalDns(
-                            blueprint_zone_type::ExternalDns {
-                                dataset,
-                                http_address: "[::1]:80".parse().unwrap(),
-                                dns_address: OmicronZoneExternalFloatingAddr {
-                                    id: ExternalIpUuid::new_v4(),
-                                    addr: SocketAddr::new(external_dns_ip, 53),
-                                },
-                                nic: NetworkInterface {
-                                    id: Uuid::new_v4(),
-                                    kind: NetworkInterfaceKind::Service {
-                                        id: external_dns_id.into_untyped_uuid(),
+                zones: [
+                    (
+                        external_dns_id,
```

Comment:
Manually duplicating the IDs seems a little dicey or easy to mess up when copy/pasting. Could we do something like

```rust
zones: [
    // all the BlueprintZoneConfigs
].into_iter().map(|z| (z.id, z)).collect()
```

instead to all the instances of this pattern? (Looks like it shows up a few times in this file and a few times in other files)

Reply:
Good idea! I should have thought of this :D

Reply:
Hah, that change shrunk the diff quite a bit.